doc/joss/paper.md

```diff
@@ -0,0 +1,168 @@
+high-performance kernels for a wide variety of parameters. 
+
+
+
+![Core architecture of Ginkgo. All solvers, preconditioners and matrix formats
+are accessible through the same LinOp interface.](figures/ginkgo-hierarchy.png)
+
+Ginkgo adopts a rigorous approach to testing and maintenance. Using continuous
+integration tools such as Gitlab-CI and Github-Actions, we make sure that 
+the library builds on a range of compilers and environments. To verify that our code
+quality is of the highest standards we use the Sonarcloud platform [@sonarcloud],
+a code analyzer to analyze and report any code smells. To ensure correctness of
+the code, we use the Google Test library [@gtest] to rigorously test each of the
+device specific kernels and the core framework of the library. 
+
+
+# Performance and Benchmarking
+
+The Ginkgo software is tailored for High Performance Computing and provides high
+performance implementations on modern manycore architectures. In particular,
+Ginkgo is competitive with hardware vendor libraries such as hipSPARSE and
+cuSPARSE [@2019spmvhip]. The two following figures highlight this fact by comparing
+the performance of Ginkgo's Hybrid and CSR SpMV kernels against the counterparts
+from the vendor libraries AMD hipSPARSE and NVIDIA cuSPARSE for all matrices
+available in the Suite Sparse Matrix Collection.
+
+![Ginkgo Hybrid spmv provides better performance than (left) cuSPARSE and
+(right) hipSPARSE](figures/ginkgo-hybrid.png) 
+
+
+![Ginkgo CSR spmv performance is competitive against (left) cuSPARSE and (b)
+hipSPARSE](figures/ginkgo-csr.png) 
+
+Ginkgo provides comprehensive logging facilities both in-house and with interfaces
+to external libraries such as PAPI [@papi]. This allows for detailed analysis of
+the kernels while reducing the intellectual overhead of optimizing the applications.
+
+To enhance reproducibility from a performance perspective, the Ginkgo project
+aims at providing Continuous Benchmarking on top of its extensive Continuous
+Integration setup [@gko_cb]. To this end, we provide the performance results of
+our kernel implementations in an open source git repository [@gko-data]. 
+A unique feature of Ginkgo is the availability of an interactive webtool, the
+Ginkgo Performance explorer [@gpe], which can plot results from the
+aforementioned data repository. Additionally, we have also put in some effort in
+making benchmarking easier, within the Ginkgo repository using the `rapidjson`
+[@rapidjson] and `gflags` [@gflags] libraries to run and generate benchmarking
+results for a variety of Ginkgo features.
+
+# Acknowledgements
+Research and software development in Ginkgo received support from the Helmholtz 
+association (Impuls und Vernetzungsfond VH-NG-1241), and the US Exascale Computing 
+Project (17-SC-20-SC), a collaborative effort of the U.S. Department of Energy Office 
+of Science and the National Nuclear Security Administration.
+
+# References
```

Comment:
The references look weird to me: The first name is written as surname, givenname, while the other authors are written as givenname surname. Is that their choice?

Reply:
I guess we do not have the option for that. They keep the format for consistency in JOSS paper. [Related Issue](https://github.com/openjournals/joss/issues/230)